### PR TITLE
`TreeView` expand chevron should not invoke item

### LINKED ISSUE #8030: [NPE] Clicking on TreeView chevron expands and closes the node instantly
### Current behavior

In [NPE canary](https://purple-pond-0688d230f.azurestaticapps.net/), clicking on the tree node expands and closes a node instantly:

20220207-treeview-expand-chevron
.

Clicking on the text next to it does not show the issue.

### Expected behavior

Tree node opens properly

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

4.1.0-dev.724

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): closes #8030

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Clicking expand chevron on `TreeView` item invokes item

## What is the new behavior?

- ✅ Clicking chevron no longer invokes item
- ⛔ Items cannot be invoked at all 👀

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
@dr1rrb Hey David, I will probably need your help with this one. The problem is probably caused by the recent pointer changes around `SelectorItem`. In `TreeViewItem` pointer press on expeand/collapse chevron is handling the pointer press:

https://github.com/unoplatform/uno/blob/c03a95aee17b6be146c1157147b943363d7207a7/src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs#L756-L763

This means the `SelectorItem` `OnPointerPressed` never gets executed:

https://github.com/unoplatform/uno/blob/c03a95aee17b6be146c1157147b943363d7207a7/src/Uno.UI/UI/Xaml/Controls/Primitives/SelectorItem.cs#L287-L295

When pointer releases, `OnPointerReleased` on `SelectorItem` is executed:

https://github.com/unoplatform/uno/blob/c03a95aee17b6be146c1157147b943363d7207a7/src/Uno.UI/UI/Xaml/Controls/Primitives/SelectorItem.cs#L306-L319

To fix the linked issue, it would be sufficient to add `&& _isPressed` to the `if` statement. Unfortunately, this will then prevent the item to be clicked directly (outside of the chevron). From my debugging it seems that when `OnPointerPressed` executes, the `if` statement fails because `CapturePointer(args.Pointer, PointerCaptureKind.Implicit))` returns `false`, because apparently the pointer is already implicitly captured. This means `_isPressed` never becomes `true` - and that means the added check in `OnPointerReleased` will prevent `OnItemClicked`.

Hope this makes sense, happy to show it in action if needed. Would you have some idea how to fix this? Does the `CapturePointer` need to be part of the condition in `OnPointerPressed`?

**Comment by jeromelaban** [comment]
This PR overlaps https://github.com/unoplatform/uno/pull/8069 from @Xiaoy312.

**Comment by dr1rrb** [comment]
@MartinZikmund I've pushed a commit which do fix the second part of your issue https://github.com/unoplatform/uno/pull/8079/commits/5000a71343d2bc5fb33fb2d8e1345fd4c4f6c766

I marked this PR as ready for review, but I let you determine if the final fix is also valid for you.

**Comment by dr1rrb** [comment]
@MartinZikmund I've pushed a commit which do fix the second part of your issue https://github.com/unoplatform/uno/pull/8079/commits/5000a71343d2bc5fb33fb2d8e1345fd4c4f6c766

I marked this PR as ready for review, but I let you determine if the final fix is also valid for you.

File: src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/TreeView/TreeView_ItemInvoked.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Microsoft.UI.Xaml.Controls;
+
+namespace UITests.Microsoft_UI_Xaml_Controls.TreeViewTests
+{
+	[Sample("TreeView")]
+    public sealed partial class TreeView_ItemInvoked : Page
```
Comment (dr1rrb) [inline src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/TreeView/TreeView_ItemInvoked.xaml.cs:21]: alignment / tabs :)

**Review by dr1rrb — APPROVED** [review]

**Pull request merged**